packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ImageAndTextViewer/components/ImageAndTextControls.js

```diff
@@ -0,0 +1,237 @@
+
+  const handleKeyDown = (event) => {
+    const index = currentFrame
+    switch (event.key) {
+      case 'ArrowRight': {
+        event.preventDefault()
+        event.stopPropagation()
+
+        handleNext()
+        if (index < locations.length - 1) {
+          document.getElementById(`thumbnail-${index + 1}`).focus()
+        } else {
+          document.getElementById(`thumbnail-${0}`).focus()
+        }
+        break
+      }
+      case 'ArrowLeft': {
+        event.preventDefault()
+        event.stopPropagation()
+
+        handlePrevious()
+        if (index > 0) {
+          document.getElementById(`thumbnail-${index - 1}`).focus()
+        } else {
+          document.getElementById(`thumbnail-${locations.length - 1}`).focus()
+        }
+        break
+      }
+      default: {
+        return true
+      }
+    }
+  }
+
+  const handlePrevious = () => {
+    if (currentFrame > 0) {
+      onFrameChange(currentFrame - 1)
+    } else {
+      onFrameChange(locations.length - 1)
+    }
+  }
+
+  const handleNext = () => {
+    if (currentFrame < locations.length - 1) {
+      onFrameChange(currentFrame + 1)
+    } else {
+      onFrameChange(0)
+    }
+  }
+
+  const handleThumbnailClick = (index) => {
+    onFrameChange(index)
+  }
+
+  return (
+    <ThemeContext.Extend value={controlsTheme}>
+      <Box background={backgrounds} ref={controlsContainer}>
+        <Box
+          fill
+          pad={smallScreenStyle ? { horizontal: '10px', vertical: '5px' } : { horizontal: '20px', vertical: '10px' }}
+          gap={smallScreenStyle ? 'xsmall' : 'small'}
+        >
+          <Box direction='row' justify='center' gap={smallScreenStyle ? 'xsmall' : 'small'}>
+            <DirectionButton
+              a11yTitle={smallScreenStyle ? t('SubjectViewer.MultiFrameViewer.FrameCarousel.previousFrameLabel') : ''}
+              icon={<FormPrevious />}
+              label={smallScreenStyle ? '' : t('SubjectViewer.MultiFrameViewer.FrameCarousel.previousFrameLabel')}
+              onClick={handlePrevious}
+            />
+            <Box
+              aria-label='Image thumbnail and text icon'
+              direction='row'
+              gap='5px'
+              role='tablist'
+              align='center'
+            >
+              {locations?.length &&
+                locations.map((location, index) => {
+                  const [mimeType] = Object.keys(locations[index])
+                  const [url] = Object.values(location)
+                  
+                  const activeFrame = currentFrame === index
+                  const tabIndex = activeFrame ? 0 : -1
+
+                  if (mimeType !== 'text/plain') {
+                    // fetching 100x100 because subject images have varying ratios and we want the image's height to be ~40px
+                    const thumbnailerUrl = `https://thumbnails.zooniverse.org/100x100${url.slice(
+                      7,
+                      url.length
+                    )}` 
```

Comment:
I have an open to-do card in the Flipbook Project Board about this handling of thumbnails. Ideally, we should write a fallback for thumbnails not hosted at `https://panoptes-uploads.zooniverse.org` because in FlipbookControls.js, the `thumbnailerUrl` is looking for `https://thumbnails.zooniverse.org/100x100/panoptes-upload...` and that will only work if the subject image is indeed a panoptes upload.

If I change handling of thumbnail buttons in flipbook controls, I'll take that opportunity to combine ImageAndTextControls's and FlipbookControls's frame navigation into a shared FrameNavigation component.

Reply:
That's a great point, I hadn't thought about subject locations hosted outside of panoptes-uploads. Plan noted sounds good to me.